core/src/main/java/org/kiwix/kiwixmobile/core/page/bookmark/viewmodel/effects/ShowDeleteBookmarksDialog.kt

```diff
@@ -38,7 +38,7 @@ data class ShowDeleteBookmarksDialog(
 ) : SideEffect<Unit> {
   @Inject lateinit var dialogShower: DialogShower
   override fun invokeWith(activity: AppCompatActivity) {
-    (activity as BookmarksActivity).activityComponent.inject(this)
+    activity.coreActivityComponent.inject(this)
```

Comment:
is this the cached component? no probably not because we don't have that on AppCompatActivity